run.py

```diff
@@ -1,10 +1,28 @@
-import os
+import os, sys
 import faulthandler
 
 faulthandler.enable()
 
-if 'CASROOT' in os.environ:
-    del os.environ['CASROOT']
+# macOS implements a security sandbox policy to executable
+# programs when they are launched from the launchd context, e.g. 
+# when double clicked from the Finder or launched from the shell
+# with the 'open' OS command.  Therefore, the macOS build requires
+# the path to the environment variable 'CSF_ShadersDirectory'
+# to be specified as an absolute path rather than a relative one.
+# 
+# The following code performs a runtime substitution of the
+# 'CSF_ShadersDirectory' environment variable if it is discovered.
+# It introspects this program's absolute path and appends the
+# relative path discovered from the environment.
+
+if sys.platform == 'darwin':
```

Comment:
this should probably go to the runtime hook , also do use Path i.s.o. manual additions.

Reply:
What is the context of the runtime hook (i.e. where is it injected at runtime?)  The code in run.py is able to introspect its path and then apply the substitution since it knows the relative location of CASROOT etc.  I'm not sure if this introspection will work in pyi_rth_occ?

Reply:
Here is some info regarding runtime hooks in pyinstaller: https://pythonhosted.org/PyInstaller/when-things-go-wrong.html#changing-runtime-behavior

Note that the environment variables you are manipulating are set here
https://github.com/CadQuery/CQ-editor/blob/1cee3f4ff90b13e1dbc54bfa22e9286c3e05b10e/pyinstaller/pyi_rth_occ.py
So for me it is quite logical to handle such things in the original file. I also noticed that you are deleting CASROOT in an else branch. I'd prefer not to change any behavior regarding other platforms.

Reply:
Thanks for the references, I will check them out and work out some changes using rth.

Deleting CASROOT was simply retaining the existing behaviour before my PR.  I wasn't sure what the purpose of the statement was, but I left the logic in place so that it would work the same way.

Reply:
Indeed, my bad. CASROOT delete is related to this #9

Reply:
Regarding RTH: you could just expand the relative to absolute paths here:
https://github.com/CadQuery/CQ-editor/blob/1cee3f4ff90b13e1dbc54bfa22e9286c3e05b10e/pyinstaller/pyi_rth_occ.py#L5

e.g. use something like (after doing `from path import Path`)
```python
env['CSF_ShadersDirectory'] = Path('oce/src/Shaders' ).abspath()
env['CSF_UnitsLexicon'] = Path('oce/src/UnitsAPI/Lexi_Expr.dat').abspath()
env['CSF_UnitsDefinition'] = Path('oce/src/UnitsAPI/Units.dat').abspath()
```

Reply:
I have already implemented roughly the same in the rth and it works well.  My next commit will include this change.